frontend/src/features/admin-form/responses/ResponsesPage/storage/worker/decryption.worker.ts

```diff
@@ -102,114 +183,86 @@ async function decryptIntoCsv(
           }
         : undefined,
     )
-    try {
-      let decryptedSubmission, submissionSecretKey
-      switch (submission.submissionType) {
-        case SubmissionType.Encrypt: {
-          const decryptedObject = formsgSdk.crypto.decrypt(secretKey, {
-            encryptedContent: submission.encryptedContent,
-            verifiedContent: submission.verifiedContent,
-            version: submission.version,
-          })
-          if (!decryptedObject) {
-            throw new Error('Invalid decryption for storage mode response')
-          }
-          decryptedSubmission = processDecryptedContent(decryptedObject)
-          break
+
+    const decryptSubmissionDataResult = await decryptSubmissionData({
+      submissionData: submission,
+      secretKey,
+    })
+
+    if (!decryptSubmissionDataResult.isSubmissionDecryptionSuccessful) {
+      csvRecord.setStatus(CsvRecordStatus.Error, 'Decryption Error')
+      csvRecord.materializeSubmissionData()
+      return csvRecord as MaterializedCsvRecord
+    }
+
+    const { decryptedResponses, mrfSubmissionSecretKey } =
+      decryptSubmissionDataResult
+
+    if (
+      // Short-circuit signature verification for multirespondent submission
+      submissionType === SubmissionType.Multirespondent ||
+      verifySignature(decryptedResponses, submissionCreated)
+    ) {
+      csvRecord.setStatus(CsvRecordStatus.Ok, 'Success')
+      csvRecord.setRecord(decryptedResponses)
+    } else {
+      csvRecord.setStatus(CsvRecordStatus.Unverified, 'Unverified')
+    }
+
+    if (downloadAttachments) {
+      // Logic to determine which key to use to decrypt attachments.
+      const attachmentDecryptionKey =
+        // If no submission secret key present, it is a storage mode form. So, use form secret key.
+        !mrfSubmissionSecretKey
+          ? secretKey
+          : // It's an mrf, but old version
+            submissionType === SubmissionType.Multirespondent &&
+              !submission.mrfVersion
+            ? secretKey
+            : mrfSubmissionSecretKey
```

Comment:
nit(readability): this nested ternary could be expressed more clearly

```suggestion
      let attachmentDecryptionKey = secretKey;
      if (
        // Only the new MRF submissions have a submission secret key
        submissionType === SubmissionType.Multirespondent &&
        submission.mrfVersion && mrfSubmissionSecretKey
      ){
        attachmentDecryptionKey = mrfSubmissionSecretKey
      }
```

Reply:
suggesting to keep it as is for this pr for a couple reasons: 
- its currently using the same code over from existing implementation (just copy-pasted). this specific logic is refactored in a subsequent stacked pr to move it to its own function, readability improvements are introduced there. 
- using `let` instead of `const` has some negatives - const ensures the `attachmentDecryptionKey` is immutable and can prevent bugs.

Reply:
fair, makes sense!